kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -96,6 +105,56 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
     ///@name Operations
     ///@{
 
+    int Check(const ModelPart& rModelPart) const override
+    {
+        KRATOS_TRY
+
+        auto lambda2_vars = GatherVariables(
+            rModelPart.Elements(), [](const AdjointExtensions& rExtensions,
+                                      std::vector<const VariableData*>& rVec) {
+                rExtensions.GetFirstDerivativesVariables(rVec);
+            });
+        auto lambda3_vars = GatherVariables(
```

Comment:
why 3 for second derivatives and  and 2 for second derivatives?

Reply:
The numbering is based on the formulation which this is developed on. Thats the only reason, so it will be easier to follow with code with the formulation.